src/utils/hooks/use-field-event.ts

```diff
@@ -0,0 +1,36 @@
+import { useContext } from "react";
+import { EventContext } from "../../components/frontend-engine/event";
+
+/**
+ * Hook that interacts with the event context provider
+ * use this hook to add/dispatch/remove event listeners
+ */
+export const useFieldEvent = () => {
+	const { eventManager } = useContext(EventContext);
+
+	const addFieldEventListener = (
+		type: string,
+		id: string,
+		listener: (this: Element, ev: Event) => unknown,
```

Comment:
do functions get auto bound to `this`?

Reply:
it doesnt appear to change where `this` is bounded to and is just meant to specify what sort of element the event listener is meant to be added to (`HTMLDivElement` in this case) 

e.g.
```
const addFieldEventListener = (
	type: string,
	id: string,
	listener: (this: string, ev: Event) => unknown,
	options?: boolean | AddEventListenerOptions
) => {
	// some code
};

myElement.addFieldEventListener("myEvent", "fieldId", function (e) {
  // `this` is a string because it is typed as such
});
```

with this, I realise we don't need to type it:
- we usually use arrow functions so `this` is meaningless
- we don't need to refer to the eventManager and are meant to interact with it through the hooks

in view of that, I have removed the typing